ACHNBrowserUI/ACHNBrowserUI/views/todayDashboard/TodayTasksSection.swift

```diff
@@ -7,70 +7,66 @@
 //
 
 import SwiftUI
+import Backend
 
 struct TodayTasksSection: View {
-    // @TODO: Check for AC Helper+ subscription
-    @State private var userHasSubscription: Bool = true
+    @EnvironmentObject private var collection: UserCollection
     
+    var isSunday: Bool {
+        Calendar.current.component(.weekday, from: Date()) == 1
+    }
     
-    var body: some View {
-        Section(header: SectionHeaderView(text: "Today's Tasks", icon: "checkmark.seal.fill")) {
-            NavigationLink(destination: Text("Event 2 Detail View")) {
-                HStack {
-                    
-                    ZStack {
-                        Circle()
-                            .foregroundColor(Color("ACBackground"))
-                        Image("icon-iron")
-                            .resizable()
-                            .aspectRatio(1, contentMode: .fit)
-                    }
-                    .frame(maxHeight: 44)
-                    
-                    ZStack {
-                        Circle()
-                            .foregroundColor(Color("ACBackground"))
-                        Image("icon-bell")
-                            .resizable()
-                            .aspectRatio(1, contentMode: .fit)
-                    }
-                    .frame(maxHeight: 44)
-                    
-                    ZStack {
-                        Circle()
-                            .foregroundColor(Color("ACBackground"))
-                        Image("icon-hardwood")
-                            .resizable()
-                            .aspectRatio(1, contentMode: .fit)
-                    }
-                    .frame(maxHeight: 44)
-                    
-                    ZStack {
-                        Circle()
-                            .foregroundColor(Color("ACBackground"))
-                        Image("icon-fossil")
-                            .resizable()
-                            .aspectRatio(1, contentMode: .fit)
-                    }
-                    .frame(maxHeight: 44)
-                    
+    // MARK: - Task Bubble
+    private func makeTaskBubble(icon: String, taskName: String) -> some View {
+        ZStack {
+            Circle()
+                .foregroundColor(Color("ACBackground"))
+            Image(icon)
+                .resizable()
+                .aspectRatio(1, contentMode: .fit)
+            if collection.dailyTasks.tasks[taskName]!.hasProgress || taskName.elementsEqual("turnip") && !self.isSunday {
```

Comment:
You should have a computed var on your view called `task: Task `that return `collection.dailyTasks.tasks[taskName]!`